### PR TITLE
Storage STG 84

### LINKED ISSUE #29739: [Storage] STG 84 release prep
- Update [Constants.DefaultSasVersion](https://github.com/Azure/azure-sdk-for-net/blob/25b8e8fe38b4cf384726a698ff77572108bcdb5a/sdk/storage/Azure.Storage.Common/src/Shared/Constants.cs#L27)
- Update [StorageVersionExtentions.LatestVersion](https://github.com/Azure/azure-sdk-for-net/blob/25b8e8fe38b4cf384726a698ff77572108bcdb5a/sdk/storage/Azure.Storage.Common/src/Shared/StorageVersionExtensions.cs#L39)
- Updated [[platform-matrix-all-versions.json](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/platform-matrix-all-versions.json#L17)](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/platform-matrix-all-versions.json#L17)
- Remove [PlaybackOnly] annotation on STG 84 unit tests.

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Storage STG 84 features.  Also resolves https://github.com/Azure/azure-sdk-for-net/issues/29739

### CODE REVIEW

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs [hunk 1]
```diff
@@ -559,6 +559,7 @@ public partial class DataLakeFileAppendOptions
         public bool? Flush { get { throw null; } set { } }
         public string LeaseId { get { throw null; } set { } }
         public System.IProgress<long> ProgressHandler { get { throw null; } set { } }
+        public Azure.Storage.UploadTransferValidationOptions TransferValidationOptions { get { throw null; } set { } }
```
Comment (weshaggard) [inline sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs:562]: This doesn't look to be in the correct place.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs:562]: @jaschrep-msft.  https://apiview.dev/Assemblies/Review/0b6337907c854077a23328f405a9cfdb/4c36ae42fb1b48b4b167881f2b25501e?diffRevisionId=3b6642b96574420da9012a005a4b5382&doc=False&diffOnly=True is a good place to discuss this.

Comment (weshaggard) [reply sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs:562]: I was only referring to what looked like a bad merge but seems to be fixed now. As for the API itself yes you can leave that up to the ArchBoard review.

**Pull request merged**